lib/models/BucketPolicy.ts

```diff
@@ -48,64 +48,77 @@ const objectActions = [
     's3:PutObjectTagging',
 ];
 
+export type BucketPolicyMetadata = {
+    Id: string;
+    Version: string;
+    Statement: {
+        Sid: string;
+        Effect: string;
+        Principal: string | { AWS: string[] };
+        Action: string | string[];
+        Resource: string | string[];
+        Condition?: { [key: string]: string | string[] };
+    }[],
+};
+
 export default class BucketPolicy {
-    _json: string;
-    _policy: any;
+    private json: string;
+    private policy: BucketPolicyMetadata | null = null;
```

Comment:
would this not be more appropriate (and consistent with the changes above)?
```suggestion
    private policy?: BucketPolicyMetadata;
```